driver-core/src/main/com/mongodb/MongoNamespace.java

```diff
@@ -51,8 +57,9 @@
     public static void checkDatabaseNameValidity(final String databaseName) {
         notNull("databaseName", databaseName);
         isTrueArgument("databaseName is not empty", !databaseName.isEmpty());
-        for (String cur : asList(" ", ".")) {
-            isTrueArgument("databaseName does not contain '" + cur + "'", !databaseName.contains(cur));
+        for (int i = 0; i < databaseName.length(); i++) {
+            isTrueArgument("databaseName does not contain '" + databaseName.charAt(i) + "'",
```

Comment:
For efficiency, invert the check and uses a Set for the prohibited characters.

Reply:
Ack